lib/mempool/mempool.js

```diff
@@ -1659,13 +1741,36 @@ class Mempool extends EventEmitter {
   isDoubleSpend(tx) {
     for (const {prevout} of tx.inputs) {
       const {hash, index} = prevout;
-      if (this.isSpent(hash, index))
+      const conflict = this.getSpent(hash, index);
+
+      if (conflict && !conflict.tx.isRBF())
         return true;
     }
 
     return false;
   }
 
+  /**
+   * Get an array of all transactions currently in the mempool that
+   * spend one or more of the same outputs as an incoming transaction.
+   * @param {TX} tx
+   * @returns {Promise} - Returns (@link MempoolEntry}[].
+   */
+
+  getConflicts(tx) {
+    const conflicts = [];
+
+    for (const {prevout} of tx.inputs) {
+      const {hash, index} = prevout;
+      const conflict = this.getSpent(hash, index);
+
+      if (conflict && conflicts.indexOf(conflict) === -1)
```

Comment:
It would be better to use a `Set` instead of an array for `conflicts` in this case, `indexOf` will need to iterate through the whole array each time looking for the `conflict`. The `Set` can be converted to an array when its being returned.

Reply:
This is good feedback as well - but the function `getConflicts()` does actually need to return an array of `MempoolEntries`. The caller, `verify()` checks the entries' fee rates and in-mempool descendants, so a `BufferSet` of hashes would just mean getting those entries later anyway, possibly in multiple places.

Reply:
I guess I could use a `BufferSet` just internally for this function, like you said to reduce search time, but still keep an array of actual `MempoolEntries` to return?

Reply:
You can use `Array.from(set)` to convert a set to an array

Reply:
updated here: https://github.com/bcoin-org/bcoin/pull/811/commits/2a2c3de12fa7e1b3e46e98d7440535ac1576ac7f